src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+        if (!outputFrame) {
+            throw Error('No matching frame found');
+        }
+        LOG && console.log('read', packetReadCount, 'packets');
+
+        this.previousTargetPTS = targetPTS;
+        return outputFrame;
+    }
+
+    /**
+     * Get the next packet from the video stream and decode it into frames. Each frame has a presentation time stamp.
+     * If we've reached the end of the stream and no more packets are available, we'll extract the last frames from
+     * the decoder and destroy it.
+     */
+    async _getNextPacketAndDecodeFrames() {
+        const packet = await this._getNextVideoStreamPacket();
+        LOG && console.log('packet pts:', packet?.pts);
+
+        // NOTE: maybe we should only decode frames when we're relatively close to our targetPTS?
+        // extract frames from the packet
+        let decodedFrames = null;
+        if (packet !== null && this.decoder) {
+            decodedFrames = await this.decoder.decode(packet as Packet);
+            LOG && console.log('decodedFrames', decodedFrames.frames.length, decodedFrames.frames.map(f => f.pts));
+        }
+        // we've reached the end of the stream
+        else {
+            if (this.decoder) {
+                LOG && console.log('getting the last frames from the decoder');
+                // flush the decoder -- this will return the last frames and destroy the decoder
+                decodedFrames = await this.decoder.flush();
+                this.decoder = null;
+            }
+            else {
+                LOG && console.log('no more frames to decode');
+            }
+        }
+
+        let frames = [];
+        if (decodedFrames && decodedFrames.frames.length !== 0) {
+            frames = decodedFrames.frames;
+        }
+
+        return { packet, frames };
+    }
+
+    async _getNextVideoStreamPacket(): Promise<null | Packet> {
+        LOG && console.log('_getNextVideoStreamPacket');
+
+        let packet = await this.demuxer.read();
+        // LOG && console.log('packet pts', packet.pts, 'stream_index', packet.stream_index);
+        while (packet && packet.stream_index !== this.streamIndex) {
+            packet = await this.demuxer.read();
+            // LOG && console.log('packet pts', packet.pts, 'stream_index', packet.stream_index);
+            if (packet === null) {
+                LOG && console.log('no more packets');
+                return null;
+            }
+        }
+        LOG && console.log('returning packet', !!packet, 'pts', packet?.pts, 'si', packet?.stream_index);
+        return packet as Packet;
+    }
+
+    _resizeFrameData(frame): Uint8ClampedArray {
+        const components = 4; // 4 components: r, g, b and a
+        const size = frame.width * frame.height * components;
+        const rawData = new Uint8ClampedArray(size); // we should probaby reuse this buffer
+        const sourceLineSize = frame.linesize as unknown as number;
+        const pixels = frame.data[0] as Uint8Array;
```

Comment:
```suggestion
        // frame.data can contain multiple "planes" in other colorspaces,
        // but in rgba, there is just one "plane", so our data is in frame.data[0]
        const pixels = frame.data[0] as Uint8Array;
```

Might be good to explain this!